package.json

```
@@ -14,6 +14,7 @@
     "ajv": "8.17.1",
     "ajv-formats": "3.0.1",
     "axios": "1.7.7",
+    "browser-version-detection": "^2.1.8",
```

```suggestion
    "browser-version-detection": "2.1.8",
```

At a minimum, we should lock this version down so it doesn't automatically change.

As an FYI, this version hasn't seen [any updates for 7 years](https://www.npmjs.com/package/browser-version-detection), and the implementation is [straightforward](https://github.com/pure-js/browser-detection/blob/main/src/detect-browser-version.ts), so I would continue to advocate for not depending on a package to do this and instead using a simple regex for browser detection (which is ultimately what this library and the previous one do).

In general, [browser detection](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) is not recommended, although I know we also do this on Terra UI. It might be better to identify the specific features that Safari doesn't support, and long term work to either stop depending on these features or use polyfills for those features that we really need. This is a separate issue from showing the alert, so in the meantime we might still want to show the alert regardless of whether we are doing browser or feature detection.

---

I'm a little confused here ... the latest release version I see is [3.1.0](https://github.com/pure-js/browser-detection/releases) that is from last year. Is there a reason we're looking at 2.1.8 (which I don't even see in the list of releases)? I do see that NPM sees [2.1.8](https://www.npmjs.com/package/browser-version-detection), which is extra confusing.

---

Aside from the library version question, I do agree in theory with Florian's feedback. To date, the only issue we've seen with Safari is variations in CSS rendering. I don't think we're using any browser-specific features, so to really tackle that question, we'd need a different kind of investigation effort.